### PR TITLE
Stateful teal test app info

### PR DESCRIPTION
**Pull request by egieseke**
## Summary

Implemented a Stateful teal app info test using expect.

## Test Plan

Ran integration tests.

### CODE REVIEW

File: test/e2e-go/cli/goal/expect/statefulTealAppReadTest.exp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+if { [catch {
+    source  goalExpectCommon.exp
+
+    puts "starting statefulTealAppReadTest"
+
+    puts "TEST_ALGO_DIR: $TEST_ALGO_DIR"
+    puts "TEST_DATA_DIR: $TEST_DATA_DIR"
+
+    statefulTealAppReadTest  $TEST_ALGO_DIR  $TEST_DATA_DIR
```
Comment (algorandskiy) [inline test/e2e-go/cli/goal/expect/statefulTealAppReadTest.exp:139]: exceed spaces around TEST_ALGO_DIR?

File: test/e2e-go/cli/goal/expect/statefulTealAppReadTest.exp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+log_user 1
+
+source  goalExpectCommon.exp
+
+set TEST_ALGO_DIR [lindex $argv 0]
+set TEST_DATA_DIR [lindex $argv 1]
+
+proc statefulTealAppReadTest { TEST_ALGO_DIR TEST_DATA_DIR} {
+
+    set TIME_STAMP [clock seconds]
```
Comment (algorandskiy) [inline test/e2e-go/cli/goal/expect/statefulTealAppReadTest.exp:13]: This looks the same as `test/scripts/e2e_subs/e2e-app-bootloader.sh` but in `expect`, what is a difference?

Comment (egieseke) [reply test/e2e-go/cli/goal/expect/statefulTealAppReadTest.exp:13]: Very similar, one difference is that the wallet is specified.

File: test/e2e-go/cli/goal/expect/statefulTealAppReadTest.exp (hunk 1 above)
Comment (egieseke) [reply test/e2e-go/cli/goal/expect/statefulTealAppReadTest.exp:139]: I will clean this up in a subsequent PR.

**Pull request merged**